File: java/src/test/java/problems1_100/ValidParenthesesTest.java
package problems1_100;

import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class ValidParenthesesTest extends SampledTest {
    @Test
    public void test() {
        for (ValidParentheses solution : ValidParentheses.values()) {
            for (ValidParenthesesSample sample : getSamples(ValidParenthesesSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.isValid(sample.input.s))
                    .isEqualTo(sample.output);
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems1_100/ValidParenthesesTest.kt
package problems1_100

import com.google.common.truth.Truth.assertWithMessage
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class ValidParenthesesTest : SampledTest() {
    @Test
    fun test() =
        ValidParentheses.entries.forEach { solution ->
            getSamples<Array<ValidParenthesesSample>>().forEach { (input, output, message) ->
                assertWithMessage(message)
                    .that(solution.isValid(input.s))
                    .isEqualTo(output)
            }
        }
}
